### PR TITLE
PLAT-1256 | verify content

### PR DESCRIPTION
**Pull request by bognix**

### CODE REVIEW

File: resources/assets/js/components/UserGeneratedContentHeader.spec.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		resolveComponent.vm.$emit('resolveResource');
+		expect(Object.keys(wrapper.emitted()).includes('resolveResource')).to.equal(true);
+
+		resolveComponent.vm.$emit('unresolveResource');
+		expect(Object.keys(wrapper.emitted()).includes('unresolveResource')).to.equal(true);
+	});
+
+	it('emits verify events correctly', () => {
+		const wrapper = createComponent({
+			propsData: { resolvable: true }
```
Comment (bognix) [inline resources/assets/js/components/UserGeneratedContentHeader.spec.js]: propsData here are redundant.

File: resources/assets/js/components/global/Delete.vue [hunk 2]
```diff
@@ -28,7 +28,7 @@
 </style>
 
 <script>
-import axios from 'axios';
+import axios from 'axios/index';
```
Comment (bognix) [inline resources/assets/js/components/global/Delete.vue]: can this change be reverted?

File: resources/assets/js/components/global/Verify.vue [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<template>
+	<p
+		v-if="$moderatorFeatures.isAllowed('access') && resource.verified_at"
```
Comment (bognix) [inline resources/assets/js/components/global/Verify.vue]: we could unify the logic, right now the logic related with displaying the resolve component is outside it, here it's inside component.

File: resources/assets/js/components/global/Verify.vue [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+<script>
+import moderator from 'js/perimeters/moderator';
+import FileVerifiedSvg from 'images/icons/file-verified.svg';
+
+export default {
+	components: {
+		FileVerifiedSvg,
+	},
+	props: ['resource'],
```
Comment (bognix) [inline resources/assets/js/components/global/Verify.vue]: props validation.

File: resources/assets/js/components/global/reactions/Vote.vue [hunk 5]
```diff
@@ -18,20 +18,26 @@
...
-		flex-direction: column
+		.icon
+			margin-right: $margin-small-minus
 
-	.vote-down
-		color: $color-red
-		flex-direction: column-reverse
+		&.vote-up
+			color: $color-green
+			border: 1px solid currentColor
```
Comment (bognix) [inline resources/assets/js/components/global/reactions/Vote.vue]: border should be in vote-down as well.

File: resources/assets/js/components/global/reactions/Vote.vue [hunk 6]
```diff
@@ -18,20 +18,26 @@
...
 		align-items: center
 		cursor: pointer
-		display: flex
+		display: inline-flex
 		font-size: $font-size-minus-2
 		font-weight: $font-weight-bold
-		padding: 0 $margin-small
 		position: relative
 		transition: all 0.2s
+		height: 27px
```
Comment (bognix) [inline resources/assets/js/components/global/reactions/Vote.vue]: maybe those styles can be cleaned up a bit.

File: resources/assets/js/components/global/reactions/Watch.vue [hunk 7]
```diff
@@ -18,13 +18,15 @@
 
 	.watch
 		align-items: center
-		border: $border-light-gray
+		border: 1px solid #A7ACBD
```
Comment (bognix) [inline resources/assets/js/components/global/reactions/Watch.vue]: this color is taken from new styleguide. How about importing colors from new styleguide to platform? @kvas-damian @rogatty @gitkuba

File: resources/assets/js/components/global/reactions/Watch.vue [hunk 8]
```diff
@@ -18,13 +18,15 @@
...
 		cursor: pointer
 		display: inline-flex
 		font-size: $font-size-minus-2
-		margin: 0 $margin-small
 		padding: 0 $margin-small
 		text-transform: uppercase
+		height: 27px
```
Comment (bognix) [inline resources/assets/js/components/global/reactions/Watch.vue]: this height is the same as for vote, if they should be the same they should be styled differently.

File: resources/assets/js/components/qna/QnaQuestion.vue [hunk 9]
```diff
@@ -1,85 +1,42 @@
...
-					<wnl-resolve
-						:resource="question"
-						@resolveResource="resolveQuestion(id)"
-						@unresolveResource="unresolveQuestion(id)"
-					/>
-				</div>
-				<slot name="context" />
+			<div class="qna-meta qna-question__header">
+				<wnl-user-generated-content-header
+					resolvable
```
Comment (bognix) [inline resources/assets/js/components/qna/QnaQuestion.vue]: probably we are missing some check here - double check.

File: resources/assets/sass/_mixins.sass [hunk 10]
```diff
@@ -102,3 +102,12 @@
 	border-radius: 0
 	box-shadow: none
 	text-align: center
+
+=action-button()
```
Comment (bognix) [inline resources/assets/sass/_mixins.sass]: I don't think we need this mixin, it's used only once.

File: resources/assets/js/components/global/reactions/Watch.vue (hunk 7 above)
Comment (rogatty) [reply resources/assets/js/components/global/reactions/Watch.vue]: How about we import them locally where needed? That's what PhpStorm suggests anyway.

File: resources/assets/js/components/global/Verify.vue [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+		&--verified
+			color: $color-green
+
+		&__svg
+			width: 16px
+			height: 16px
+
+		&:hover
+			color: $color-red
```
Comment (kvas-damian) [inline resources/assets/js/components/global/Verify.vue]: I think it should be disabled for non-moderator users

File: resources/assets/js/components/global/Verify.vue [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+</template>
+
+<style lang="sass" rel="stylesheet/sass" scoped>
+	@import 'resources/assets/sass/variables'
+
+	.verify
+		font-size: $font-size-minus-1
+		cursor: pointer
+		width: auto
+		color: $color-gray
```
Comment (kvas-damian) [inline resources/assets/js/components/global/Verify.vue]: I think it should be disabled for non-moderator users

File: resources/assets/js/components/global/Verify.vue (hunk 11 above)
Comment (kvas-damian) [reply resources/assets/js/components/global/Verify.vue]: Did Luk accept this color?

File: resources/assets/js/components/global/Verify.vue [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+			width: 16px
+			height: 16px
+
+		&:hover
+			color: $color-red
+
+		span
+			margin-left: $margin-small
+
+		i
```
Comment (kvas-damian) [inline resources/assets/js/components/global/Verify.vue]: to be removed

File: resources/assets/js/components/UserGeneratedContentHeader.spec.js [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+			sync: false
+		});
+	};
+
+	it('renders correctly for regular user', () => {
+		const wrapper = createComponent();
+		expect(wrapper.findAll('#avatar').length).to.equal(1);
+		expect(wrapper.findAll('#resolve').length).to.equal(0);
+		expect(wrapper.findAll('#delete').length).to.equal(0);
+		expect(wrapper.findAll('#verify').length).to.equal(1);
```
Comment (kvas-damian) [inline resources/assets/js/components/UserGeneratedContentHeader.spec.js]: Something wrong is happening here. If it's a regular user, we should not see the `verify` component unless you pass `verified_at` property.

File: resources/assets/js/components/global/Verify.vue [hunk 15]
```diff
@@ -0,0 +1,77 @@
+<template>
+	<p
+		v-if="showBadge"
+		:class="['verify', 'icon', resource.verified_at && 'verify--verified']"
```
Comment (kvas-damian) [inline resources/assets/js/components/global/Verify.vue]: use `verified` computed

File: resources/assets/js/components/global/reactions/Vote.vue [hunk 16]
```diff
@@ -1,37 +1,30 @@
...
-		@click="toggleReaction"
-	>
-		<span class="icon is-small">
-			<i class="fa" :class="hasReactedClass" />
-		</span>
-		<span class="count">{{count}}</span>
-		<span class="flash-bg" :class="{'flash': wasJustClicked}" />
-	</div>
+	<wnl-rectangle-button :class="iconClass" @click="toggleReaction">
+		<div>
```
Comment (kvas-damian) [inline resources/assets/js/components/global/reactions/Vote.vue]: do we need this div?

File: resources/assets/js/components/qna/QnaAnswer.vue [hunk 17]
```diff
@@ -108,26 +81,18 @@
...
-
-import { timeFromS } from 'js/utils/time';
 
 export default {
 	name: 'QnaAnswer',
 	components: {
-		'wnl-avatar': Avatar,
-		'wnl-delete': Delete,
+		WnlUserGeneratedContentHeader,
 		'wnl-vote': Vote,
```
Comment (kvas-damian) [inline resources/assets/js/components/qna/QnaAnswer.vue]: Please migrate component imports to `Wnl...` pattern

File: resources/assets/js/store/modules/comments.js [hunk 18]
```diff
@@ -171,13 +156,41 @@ export const commentsActions = {
...
+		const { data: updatedComment } = await _updateComment(payload.id, { verified: true });
+		commit(types.UPDATE_COMMENT, {
+			id: payload.id,
+			resource: payload.commentableResource,
+			resourceId: payload.commentableId,
+			updatedComment
+		});
+	},
+	async unverifyComment({ commit }, payload) {
+		const { data: updatedComment } = await _updateComment(payload.id, { verified: false });
```
Comment (kvas-damian) [inline resources/assets/js/store/modules/comments.js]: Add error handling

File: resources/assets/js/components/UserGeneratedContentHeader.spec.js (hunk 14 above)
Comment (kvas-damian) [reply resources/assets/js/components/UserGeneratedContentHeader.spec.js]: Discussed off-line: This component decides if it should render anything or not inside the component - so it's always present.

**Pull request merged**